src/lib/storage/column_encoders/utils.hpp

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <memory>
+
+#include "all_type_variant.hpp"
+#include "storage/encoded_columns/column_encoding_type.hpp"
+
+namespace opossum {
+
+class BaseColumn;
+class BaseColumnEncoder;
+class BaseValueColumn;
+
+/**
+ * @brief Creates an encoder by encoding type
+ */
+std::unique_ptr<BaseColumnEncoder> create_encoder(EncodingType encoding_type);
```

Comment:
Do we have to put this into the global namespace, especially if there is only one user? How about `BaseColumnEncoder::create_encoder`?

Reply:
Mhm, should a base column know about all of its sub classes?

Reply:
I don’t know. I don’t see a problem here. This file is only included in a cpp-file and isn’t in the global namespace except in this one file that includes it.